gtda/local_homology/simplicial.py

```diff
@@ -0,0 +1,411 @@
+        # points lie in the 'neighborhouds' of each points. The points outside
+        # the 'neighborhoud' defined by the largest entry are discarded, and
+        # the points between the smaller and largest 'neighborhouds' are 'coned
+        # off'. See more in the corresponding fit methods.
+        self.neighborhoud_param = neighborhoud_param
+
+        # make sure the neighborhoud_param has been set correctly.
+        if self.neighborhoud_param[0] > self.neighborhoud_param[1]:
+            warnings.warn('First neighborhoud_param should be smaller than second.\
+                The values are permuted')
+            self.neighborhoud_param = (self.neighborhoud_param[1],
+                                       self.neighborhoud_param[0])
+        if self.neighborhoud_param[1] == 0:
+            warnings.warn('Second neighborhoud_param has to be strictly greater than 0.\
+                Second radius set to 1.')
+            self.radii = (self.radii[0], 1)
+        if self.neighborhoud_param[0] == self.neighborhoud_param[1]:
+            warnings.warn('For meaningfull features, the first neighborhoud_param should\
+                be strictly smaller than second.')
+
+    def fit(self, X, y=None):
+        return self
+
+    def transform(self, X):
+        """
+        Computes the local persistence diagrams at each element of X, and
+        returns a list of persistence diagrams, indexed as the points of X.
+        This is done in several steps:
+            - First compute the nearest neighbors in the point cloud that
+            was fitted on, for both values in n_neighbors.
+            - For each point, compute the relevant points (corresponding to
+            the larger neighborhoud_param value), the close points
+            (corresponding to the smaller neighborhoud_param value), and the
+            annulus to cone off (relevant points, but not close points).
+            Compute the distance matrix of the relevant points, and add an
+            additional row and column corresponding to the coning off point.
+            - Finally compute the persistence diagrams of each coned matrices.
+
+        Parameters
+        ----------
+
+        X : ndarray of shape (n_points, dimension)
+             Input data representing  point cloud:
+             an array of shape ``(n_points, n_dimensions)``.
+
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        Xt : ndarray of shape (n_samples, n_features, 3)
+            Array of persistence diagrams computed from the feature arrays.
+            ``n_features`` equals :math:`\\sum_q n_q`, where :math:`n_q`
+            is the maximum number of topological features in dimension
+            :math:`q` across all samples in `X`.
+
+        """
+
+        check_is_fitted(self)
+        check_array(X, accept_sparse=False)
```

Comment:
```suggestion
        Xt = check_array(X, accept_sparse=False)
```

because this way we make sure the converted and validated array is used from now on. (If you accept this, please replace all instances of `X` with `Xt` below also.)

Reply:
Only caveat is that since (I am assuming) we want to return Xt at the end of the method, this forces us to have two different variables with the same name Xt (See line 158). Is that ok?

Reply:
Are you worried that
`        Xt = self.homology.fit_transform(coned_mats)`
means we replace the previous `Xt`?  If so, that's ok!

Reply:
Ok, thanks!